tests/unit/test_dns_client.cpp

```diff
@@ -804,6 +805,16 @@ void TestDnsClient(void)
         VerifyOrQuit(addresses.Contains(sAddressInfo.mHostAddresses[index]));
     }
 
+    //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+    // Validate DNS Client `ResolveAddress()` for an invalid (non-existing) name
+
+    sAddressInfo.Reset();
+    Log("ResolveAddress(%s)", kNonExistingName);
+    SuccessOrQuit(dnsClient->ResolveAddress(kNonExistingName, AddressCallback, sInstance));
+    AdvanceTime(100);
+    VerifyOrQuit(sAddressInfo.mCallbackCount == 1);
+    VerifyOrQuit(sAddressInfo.mError == kErrorNotFound);
```

Comment:
Is there a way to check here that the question-count in the answer is 1 ? This could be done for the valid name query and the invalid name query.

Reply:
No, I don't think there's an easy way to validate this in the unit test, as the parsed error code and/or the parsed result (e.g., IPv6 addresses) are provided in the callback. It's not worth adding new code and machinery just to test this.

Reply:
Right, agree that no code needs to be added only for that!